### PR TITLE
Move to float32

### PR DESCRIPTION
**Pull request by oHunewald**

### CODE REVIEW

File: test/parallel.jl [hunk 1]
```diff
@@ -36,7 +72,10 @@ som2 = initGigaSOM(df_som, 10, 10)
 
 end
 
-@time som2 = trainGigaSOM(som2, df_som, epochs = 2, r = 6.0)
+@time som2 = trainGigaSOM(som2, df_som, epochs = 10)
+
+mywinners = mapToSOM(som2, df_som)
```
Comment (laurentheirendt) [inline test/parallel.jl]: This should be `mapToGigaSOM`.

File: test/parallel.jl [hunk 2]
```diff
@@ -11,12 +11,48 @@ cc = map(Symbol, lineage_markers)
...
+        @test typeof(dm) == Array{Float32, 2}
+
+    end
+
+    @testset "gaussianKernel" begin
+
+        y = GigaSOM.gaussianKernel(Float32(10.0), Float32(6.0))
+        @test typeof(y) == Float32
+    end
+
```
Comment (laurentheirendt) [inline test/parallel.jl]: May you remove all these empty lines?

File: test/parallel.jl [hunk 3]
```diff
@@ -11,12 +11,48 @@ cc = map(Symbol, lineage_markers)
 df_som = daf.fcstable[:,cc]
 
 # concatenate the dataset for performance testing
+# df_som = vcat(df_som, df_som)
```
Comment (laurentheirendt) [inline test/parallel.jl]: Feel free to delete these lines .... 😄

File: src/types.jl [hunk 4]
```diff
@@ -111,7 +111,10 @@ Return Gaussian(x) for μ=0.0 and σ = r/3.
 (a value of σ = r/3 makes the training results comparable between different kernels
 for same values of r).
 """
-function gaussianKernel(x::Float64, r::Float64)::Float64
+function gaussianKernel(x::Float32, r::Float32)::Float32
+    # dist = Distributions.pdf.(Distributions.Normal(0.0,r/3), x)
```
Comment (laurentheirendt) [inline src/types.jl]: No need to keep these lines in.

File: src/types.jl [hunk 5]
```diff
@@ -111,7 +111,10 @@ Return Gaussian(x) for μ=0.0 and σ = r/3.
 (a value of σ = r/3 makes the training results comparable between different kernels
 for same values of r).
 """
-function gaussianKernel(x::Float64, r::Float64)::Float64
+function gaussianKernel(x::Float32, r::Float32)::Float32
```
Comment (laurentheirendt) [inline src/types.jl:114]: the function header also has to be adapted with `Float32`

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: src/parallel_som.jl [hunk 6]
```diff
@@ -172,14 +171,17 @@ epoch.
 - `toroidal`: if true, the SOM is toroidal.
```
Comment (laurentheirendt) [inline src/parallel_som.jl:170]: This header needs updating.

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: src/parallel_som.jl (hunk 6 above)
Comment (oHunewald) [reply src/parallel_som.jl:170]: done

**Review by laurentheirendt — APPROVED** [review]

**Comment by laurentheirendt** [comment]
Well done @oHunewald 👍

**Comment by laurentheirendt** [comment]
Note: Function headers still need updating.

**Pull request merged**